gluten-core/src/main/scala/io/glutenproject/extension/ExpandFallbackPolicy.scala

```diff
@@ -236,19 +235,9 @@ case class ExpandFallbackPolicy(isAdaptiveContext: Boolean, originalPlan: SparkP
   }
 
   private def fallbackToRowBasedPlan(outputsColumnar: Boolean): SparkPlan = {
-    val transformPostOverrides = TransformPostOverrides(isAdaptiveContext)
+    val transformPostOverrides = TransformPostOverrides()
     val planWithTransitions = InsertTransitions.insertTransitions(originalPlan, outputsColumnar)
-    planWithTransitions.transform {
-      case c2r @ ColumnarToRowExec(_: ShuffleQueryStageExec) =>
-        transformPostOverrides.transformColumnarToRowExec(c2r)
-      case c2r @ ColumnarToRowExec(_: AQEShuffleReadExec) =>
-        transformPostOverrides.transformColumnarToRowExec(c2r)
-      // `InMemoryTableScanExec` itself supports columnar to row
-      case ColumnarToRowExec(child: SparkPlan) if PlanUtil.isGlutenTableCache(child) =>
-        child
```

Comment:
We should apply `RemoveGlutenTableCacheColumnarToRow` here manually since we sperate `TransformPostOverrides` into 3 rules.
BTW, some tests which I added in `VeloxColumnarCacheSuite` are used for Spark3.5 so may not work for now...

Reply:
I assume `RemoveGlutenTableCacheColumnarToRow` would eventually be applied? By looking into this code:

https://github.com/oap-project/gluten/blob/67978a666835dd83989cecd4168cf67276b723f1/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L832-L847

`fallbackPolicy()` would apply before `postOverrides()` in which we populate ``RemoveGlutenTableCacheColumnarToRow` rule is applied.

Reply:
Aha I see. The fallback plan is not doing `postOverrides()`.  It was my mistake.

Reply:
It won't be applied if the plan is fallback.. we should put it into `finallyRules`  if we want to apply one rule finally.

Reply:
> It won't be applied if the plan is fallback.. we should put it into `finallyRules` if we want to apply one rule finally.

Yes that's just what I am trying to do. PR updated, please take another look. Thanks.